src/main/scala/firrtl/transforms/ConstantPropagation.scala

```diff
@@ -44,7 +44,19 @@ class ConstantPropagation extends Transform {
     }
   }
 
-  object FoldAND extends FoldLogicalOp {
+  object FoldADD extends FoldCommutativeOp {
+    def fold(c1: Literal, c2: Literal) = (c1, c2) match {
+      case (_: UIntLiteral, _: UIntLiteral) => UIntLiteral(c1.value + c2.value, (c1.width max c2.width) + IntWidth(1))
```

Comment:
Is addition defined to be non-expanding when one of the arguments has width 0?  i.e. should the width be something like

    if (c1.width == 0) c2.width
    else if (c2.width == 0) c1.width
    else (c1.width + c2.width) + 1

Reply:
I'm not sure why `simplify` is padded but `fold` is not. @aswaterman wrote the code 2 years ago, do you happen to remember? https://github.com/freechipsproject/firrtl/commit/359021cd1753cd8ad6e5315da6ef2638c5000323

Reply:
Just because it was easy to make all the `fold` cases return the correct width, but that seemed too onerous for all the `simplify` cases.

Reply:
(I don't actually remember; that's just my best guess)

Reply:
Do we need to address this since we can't create zero-width literals?

Reply:
For the record, my feeling is that adding a zero-width wire should increase the width. Semantically its weird for constant prop to affect the width of a wire, unless that is properly handled (which could be a future optimization).